praw/objects.py

```diff
@@ -1899,6 +1902,34 @@ def remove_editor(self, username, *args, **kwargs):
         return self.add_editor(username=username, _delete=True, *args,
                                **kwargs)
 
+    def revisions(self, **kwargs):
+        """Return a dictionary of all page revisions.
+
+        Additional parameters are passed to
+        :meth:`~praw.__init__.BaseReddit.request_json` to be used as
+        URL parameters.
+        """
+        url = self.reddit_session.config['wiki_page_revisions'].format(
+                subreddit=six.text_type(self.subreddit),
+                page=self.page.lower())
+
+        prev_use_oauth = self.reddit_session._use_oauth
+        self.reddit_session._use_oauth = True
+        response = self.reddit_session.request_json(url, params=kwargs)
+        self.reddit_session._use_oauth = prev_use_oauth
+
+        return response['data']
+
+    def revision(self, version):
+        """Return a specific version of the WikiPage.
+
+        :param version: The version GUID of the page (returned by
+        :meth:`revisions`)
+        """
+        result = WikiPage(self.reddit_session, self.subreddit, self.page,
```

Comment:
Revisions are a list of JSON dictionaries. Each dictionary entry (a WikiRevision) has an id (a UUID), the page name itself (always the same), the reason for the change (text, may be null) the timestamp (in the usual UTC seconds format), and the author, which is a full t2 `Redditor` object. 

The revision, as far as I can tell, is really a JSON view of [WikiRevision](https://github.com/reddit/reddit/blob/master/r2/r2/models/wiki.py#L115).

It should probably become an object of its own (and the revisions list should be a `Listing` of `WikiRevision`), but I have no idea where to start with that.